### PR TITLE
LPS-193854 - Add Service Account and OAuth2 Application for Remote Publications

### PR DESCRIPTION
**Pull request by david-truong**

### CODE REVIEW

**Comment by david-truong** [comment]
ci:test:sf

**Comment by david-truong** [comment]
@xbrianlee can you review the tests and have someone on QA add some new tests for the new option.

I'll add the AC to the ticket

**Comment by david-truong** [comment]
@xbrianlee @suilin i added the AC in: https://liferay.atlassian.net/browse/LPS-193854

**Comment by david-truong** [comment]
ci:test:publications

**Comment by suilin** [comment]
Hi @david-truong, I finished analyzing the test results, there are two bugs caused by the current commits, can you look at them? thanks.

- Enabling publications via instance setting doesn't work.
1. Navigate to instance setting > Publications
2. Check to Enable checkbox and save
Result: The Enable checkbox can be checked but it doesn't work.

- Publications configurations can immigrate from one instance to another one
1. Enable Publications
2. Add a new instance
3. Add a new publication
4. Login with the new instance
Result: Publications is enabled in the new instance. 

cc @xbrianlee

**Comment by david-truong** [comment]
ci:test:publications

**Comment by david-truong** [comment]
ci:test:relevant

**Comment by david-truong** [comment]
@suilin can you take a look at the new results

**Comment by xbrianlee** [comment]
@david-truong I'll take a look at the results tomorrow and start working on the functional test since Linda's out

File: modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/global_settings.jspf [hunk 1]
```diff
@@ -61,4 +92,10 @@
 		'<portlet:namespace />unapprovedChanges',
 		<%= publicationsConfigurationDisplayContext.isPublicationsEnabled() ? false: null %>
 	);
+
+	Liferay.Util.toggleBoxes(
+		'<portlet:namespace />enablePublications',
+		'<portlet:namespace />manageRemotely',
+		<%= publicationsConfigurationDisplayContext.isPublicationsEnabled() ? false: null %>
```
Comment (pei-jung) [inline modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/global_settings.jspf:99]: sf: missing space after `false`

**Review by pei-jung — COMMENTED** [review]
@david-truong From what I can tell with the current setup, the remote publications will only work if you enable it in the publications settings page. Maybe we can add a `remotePublicationsEnabled` property to the `CTSettingsConfiguration` so that the enable/disable logic in `RemotePublicationsHelper` can be handled in a ConfigurationModelListener, which is triggered every time the property value changes. That also allows the remote publications feature to be enabled via a config file or in Instance Settings.

File: modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/global_settings.jspf (hunk 1 above)
Comment (david-truong) [reply modules/apps/change-tracking/change-tracking-web/src/main/resources/META-INF/resources/publications/global_settings.jspf:99]: @pei-jung I originally had it as a boolean but then when it was disabled, I had no way to fetch or delete the OAuth2Application without the id to use for displaying the clientId and clientSecret and cleaning it up if disabled

**Comment by xbrianlee** [comment]
@david-truong I've verified the AC in the ticket and the test results above look good. One minor thing - we're missing a space after `Client ID:`.
Screenshot 2023-08-24 at 2 03 17 PM

From the QA perspective, this PR can be forwarded. I'm working on a new functional test that covers all of the AC which I can send separately.

**Comment by xbrianlee** [comment]
ci:test:publications

**Comment by xbrianlee** [comment]
ci:test:relevant

**Comment by xbrianlee** [comment]
ci:test:sf

**Comment by xbrianlee** [comment]
@pei-jung @david-truong Is it ok to forward this PR if the tests look good? We're planning to send the test separately

**Review by pei-jung — APPROVED** [review]
LGTM! 👍

**Comment by xbrianlee** [comment]
ci:test:publications

**Comment by xbrianlee** [comment]
ci:test:relevant

**Comment by xbrianlee** [comment]
ci:forward:force

**Pull request closed without merging**